server/config/validate-config-keys.js

```diff
@@ -46,7 +46,9 @@ const sharedConfig = parseConfig( '_shared.json' );
  * invalid. Such a missing value could and likely would
  * cause runtime errors in Calypso
  */
-environmentKeys.forEach( ( [ filename, keys ] ) => {
```

Comment:
Could you try https://github.com/Automattic/wp-calypso/pull/14631 to see if the problem is solved? It uses [husky](https://github.com/typicode/husky) to set up Git hooks, and it seems to be `nvm`-aware.